app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -280,4 +273,36 @@ class MoneyActivity : SimpleActivity(), Calculator , LocationListener {
         })
         tipDialog.show()
     }
+
+    private fun displayConversionDialog(){
+        val conversionDialogBuilder = AlertDialog.Builder(this)
+        val conversionDialogView = layoutInflater.inflate(R.layout.conversion_modal, null)
+        conversionDialogView.findViewById<Button>(R.id.btn_conversion).setOnClickListener {
+            performConversion(conversionDialogView.findViewById<Spinner>(R.id.conversion_spinner1).selectedItem,
+                    conversionDialogView.findViewById<Spinner>(R.id.conversion_spinner2).selectedItem)
+            conversionDialog!!.dismiss()
+        }
+        conversionDialogBuilder!!.setView(conversionDialogView)
+        conversionDialogBuilder!!.setCancelable(true)
+        conversionDialog = conversionDialogBuilder!!.create()
+        conversionDialog!!.show()
+    }
+
+    private fun performConversion(from:Any, to:Any) {
+        if (!from.toString().equals(to.toString()) && isOnline()) {
+            calc.calcCurrency(from.toString(),to.toString(),this)
+        }else{
+            displayToast("It seems there has been a connection problem, contact you ISP for more details !")
+        }
+    }
+
+    fun isOnline(): Boolean {
```

Comment:
If this is a function that is not called elsewhere than this class then we should consider making it private to enforce method encapsulation

Reply:
I agree